### PR TITLE
When generating service mappings cast pair to set to avoid duplicate pairs

### PR DESCRIPTION
**Pull request by suddrey-qut**
Fixes issue #257

Recreation of PR #259 to do master->master

### CODE REVIEW

File: ros1_bridge/__init__.py [hunk 1]
```diff
@@ -562,17 +562,23 @@ def determine_common_services(
     for rule in mapping_rules:
         for ros1_srv in ros1_srvs:
             for ros2_srv in ros2_srvs:
+
+                pair = (ros1_srv, ros2_srv)
+
+                if pair in pairs:
+                    continue
+
```
Comment (peterpena) [inline ros1_bridge/__init__.py:570]: nitpick: eliminate the newlines in line 565, 567, 570. Otherwise LGTM.

Comment (suddrey-qut) [reply ros1_bridge/__init__.py]: Done

**Comment by dirk-thomas** [comment]
Thanks for the patch.

**Comment by peterpena** [comment]
Should this be backported?

**Comment by dirk-thomas** [comment]
> Should this be backported?

Into which ROS distro?

**Comment by peterpena** [comment]
@dirk-thomas I am not sure what ROS distro it will be. The bug seems to be introduced all the way back to Ardent. What is the protocol to decide the ROS distro?

**Comment by dirk-thomas** [comment]
> What is the protocol to decide the ROS distro?

Only active distro can be patched (so the earliest would be Dashing). Another criteria is if it is necessary / severe enough. Based on the original issue it seems trivial to avoid this case by not defining a custom mapping which replicates the default behavior. So I would think the need to backport is fairly low.

**Comment by peterpena** [comment]
I see. Thanks for the explanation. I felt the need to ask if it needed to be backported, but based on what you said, it seems that it might not be necessary.

**Pull request merged**